### ISSUE TITLE
group-weight in config.yml 

### ISSUE THREAD

**Issue by voodootje0** [lucko/LuckPerms#438]
This is my config.yml
```
group-weight:
  immortal: 45
```

But when I type /lp group immortal info it shows 25 (since that's the global weight). Also in /lp listgroups it shows 25 instead of 45.

**Comment by BrainStone** [comment]
You probably have set the weight to 25 manually.

**Comment by lucko** [comment]
Any weights set with the `setweight` command will override what's in the config.

**Comment by voodootje0** [comment]
What's then the meaning of `group-weight` in the config? There is also no explanation in the wiki.

**Comment by lucko** [comment]
The `group-weight` config option was used before the command existed. There's no real use for it now, as it's much easier to just use the commands.

**Comment by voodootje0** [comment]
Ahaaa.

Then I would suggest making the `group-weight` config option overriding the setweight command. This would be really helpful if you're having the same rank name on different servers.

**Comment by lucko** [comment]
This would break compat, and I prefer it the other way around.

**Comment by voodootje0** [comment]
Could you then add /lp group setweight <weight> [context...]?

**Comment by lucko** [comment]
Uhh, this is going to be difficult.

The weight of group is used throughout the permission resolution process. At this stage, the current context of the group is not known, therefore, it is not possible to determine a weight in a specific context.

Therefore, I am going to close this issue. I initially re-opened it because I thought changing this would be possible - it's only now when I went to actually implement it that I realised it won't be possible, due to the nature and current uses of the weight system.

Hopefully that makes sense. :)

> This would be really helpful if you're having the same rank name on different servers.

If the ranks have different 'priorities' on different servers, then that's probably a sign that they should have distinct names.
